### ISSUE TITLE
Retrieve the domain name from the Site instead of the request

### ISSUE THREAD

**Issue by skorokithakis** [pennersr/django-allauth#1840]
The request object is often unreliable (for example when the server is behind a reverse proxy). The `current_site` is much more reliable, so allauth should look there first. Is there maybe already a setting that can do this? Facebook auth doesn't work for me because allauth thinks my site's domain is `localhost`, even though it's set properly in the Site instance.

**Comment by skorokithakis** [comment]
This ticket may be invalid, sorry. I tried again and it works now, for some reason.

**Comment by undernewmanagement** [comment]
@skorokithakis actually, I also see this issue when sending the password reset emails. I have a reverse proxy setup as well.

[kong] -> [docker] -> [django] 

and my password reset urls looks like this:

https://localhost:8102/accounts/password/reset/key/1-4q2-69d52f335024013a306e/

even though I have SITE_ID = 1 in my settings.py and configured in the database.

Any insights on how to solve this are welcome. I really do not want to override views just to solve a simple configuration problem. (I just dont know where the problem is, exactly :-)

**Comment by skorokithakis** [comment]
Your reverse proxy needs to set the header. For nginx:

~~~
 proxy_set_header Host            $host;
proxy_set_header X-Forwarded-For $remote_addr;
~~~

This is clearly a bad default on the part of django-allauth, though, and I think it should be fixed.

**Comment by pennersr** [comment]
No, if you put it behind a proxy it is your responsibility to properly hand over the required HTTP headers.

**Comment by skorokithakis** [comment]
@pennersr The point of this ticket is that ignoring the `SITE_ID` is incorrect behaviour.

**Comment by pennersr** [comment]
But, regardless of the sites, any app using request.build_absolute_uri() will fail if you don't setup the proxy headers properly. So, fixing those is a must anyway.

Re-opening, because you are formally right in the sense that Django's Site.objects.get_current(request) is also ignoring the request if SITE_ID is  set.

**Comment by skorokithakis** [comment]
There's more to this than the request headers. For example, a site that uses one domain per user might not want notification emails showing "https://user.domain.com" but "https://www.domain.com" instead. The canonical domain is the one in the Sites model, not the header that's coming in.

**Comment by steverecio** [comment]
Has anyone looked into this any further? Looks like the allauth password reset emails are still vulnerable to host header injection. If an attacker is able to inject a malicious host header, then the they can control the domain for the password reset page. In theory, this could lead to phishing for current passwords of unsuspecting users.

**Comment by steverecio** [comment]
I solved it in my app by hardcoding the host header in my nginx proxy (`proxy_set_header Host            "https://mydomain.com";`) but it would be nice to offer a config variable that overrides the use of `request.build_absolute_uri()`. This would make it easier to configure the default host for my staging environment (nginx is more challenging to template per environment).

**Comment by skorokithakis** [comment]
Wow, this is a blast from the past. I'm fine with closing, yep!

**Comment by derek-adair** [comment]
ha, exactly... just doing some triage here to help focus the issue tracker and make progress on high value features!!

Cheers!

**Comment by derek-adair** [comment]
If you or anyone wants to submit this refactor w/ tests i'm sure it'd get merged.

**Comment by pennersr** [comment]
https://codeberg.org/allauth/django-allauth/issues/1840
